Sources/EdgeNetworkHandlers/EdgeNetworkService.swift

```diff
@@ -110,34 +114,39 @@ class EdgeNetworkService {
                                        streaming: requestBody.meta?.konductorConfig?.streaming,
                                        responseCallback: responseCallback)
                     responseCallback.onComplete()
-                    completion(true) // successful request, return true
+                    completion(true, 0.0) // successful request, return true
                 } else if responseCode == HttpResponseCodes.noContent.rawValue {
                     // Successful collect requests do not return content
                     Log.debug(label: self.LOG_TAG, "doRequest - Collect connection to Experience Edge was successful.")
                     responseCallback.onComplete()
-                    completion(true) // successful request, return true
+                    completion(true, 0.0) // successful request, return true
                 } else if self.recoverableNetworkErrorCodes.contains(responseCode) {
                     Log.debug(label: self.LOG_TAG, "doRequest - Connection to Experience Edge returned recoverable error code \(responseCode)")
-                    completion(false) // failed, but recoverable so retry
+                    let retryHeader = connection.responseHttpHeader(forKey: EdgeConstants.NetworkKeys.HEADER_KEY_RETRY_AFTER)
+                    var retryInterval = EdgeConstants.NetworkKeys.RETRY_INTERVAL
+                    if let retryHeader = retryHeader, let retryAfterInterval = TimeInterval(retryHeader) {
```

Comment:
It seems like the Retry-After header can be: 
```
<http-date>
A date after which to retry. See the Date header for more details on the HTTP date format.
<delay-seconds>
A non-negative decimal integer indicating the seconds to delay after the response is received.
```
As discussed with the service team, this will be in int format, but let's clarify that in a comment here and a test case that we don't fail when date is returned.

Ref https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After

Reply:
I don't see any need to add a test case for a scenario that will not be returned by the server and one that we explicitly do not support. I've added a clarifying comment to the parsing of the header.